modifiers/allocation/modifier.py

```diff
@@ -0,0 +1,415 @@
+
+        if v.timeout:
+            sbatch_opts.append(f"--time {v.timeout}")
+
+        sbatch_directives = list(f"#SBATCH {x}" for x in (srun_opts + sbatch_opts))
+
+        v.mpi_command = f"srun {' '.join(srun_opts)}"
+        v.batch_submit = "sbatch {execute_experiment}"
+        v.allocation_directives = "\n".join(sbatch_directives)
+
+    def gpus_as_gpus_per_rank(self, v):
+        """Some systems don't have a mechanism for directly requesting a
+        total number of GPUs: they just have an option that specifies how
+        many GPUs are required for each rank.
+        """
+        # This error message can come up in multiple scenarios, so pre
+        # define it if it's needed (it might not be true except where the
+        # error is raised)
+        err_msg = (
+            f"Cannot express GPUs ({v.n_gpus}) as an integer "
+            f"multiple of ranks ({v.n_ranks})"
+        )
+
+        if v.n_gpus >= v.n_ranks:
+            quotient, remainder = divide_into(v.n_gpus, v.n_ranks)
+            if remainder == 0:
+                return quotient
+            else:
+                raise ValueError(err_msg)
+        else:
+            raise ValueError(err_msg)
+
+    def flux_instructions(self, v):
+        cmd_opts = []
+        batch_opts = []
+
+        if v.n_ranks:
+            cmd_opts.append(f"-n {v.n_ranks}")
+        if v.n_nodes:
+            cmd_opts.append(f"-N {v.n_nodes}")
+        if v.n_gpus:
+            gpus_per_rank = self.gpus_as_gpus_per_rank(v)
+            cmd_opts.append(f"--gpus-per-task={gpus_per_rank}")
+
+        if v.timeout:
+            batch_opts.append("-t {v.timeout}m")
+
+        batch_directives = list(f"# flux: {x}" for x in (cmd_opts + batch_opts))
+
+        v.mpi_command = f"flux run {' '.join(cmd_opts)}"
+        v.batch_submit = "flux batch {execute_experiment}"
+        v.allocation_directives = "\n".join(batch_directives)
+
+    def mpi_instructions(self, v):
+        v.mpi_command = f"mpirun -n {v.n_ranks} --oversubscribe"
+        v.batch_submit = "{execute_experiment}"
+        v.allocation_directives = ""
+
+    def lsf_instructions(self, v):
+        cmd_opts = []
+        batch_opts = []
+
+        if v.n_ranks:
+            cmd_opts.append(f"-n {v.n_ranks}")
+        if v.n_nodes:
+            batch_opts.append(f"-nnodes {v.n_nodes}")
+        if v.n_gpus:
+            gpus_per_rank = self.gpus_as_gpus_per_rank(v)
+            batch_opts.append(f"-g {gpus_per_rank}")
+        if v.n_ranks_per_node:
+            cmd_opts.append(f"-T {v.n_ranks_per_node}")
+        # TODO: this might have to be an option on the batch_submit vs.
+        # a batch directive
+        if v.queue:
+            batch_opts.append(f"-q {v.queue}")
+        if v.timeout:
+            batch_opts.append(f"-W {TimeFormat.as_hhmm(v.timeout)}")
+
+        batch_directives = list(f"#BSUB {x}" for x in batch_opts)
+
+        v.mpi_command = f"lrun {' '.join(cmd_opts)}"
```

Comment:
Unfortunately `lrun` is a Livermore-Specific tool. Under the covers, it runs `jsrun`, which should work on all IBM Spectrum LSF-scheduled machines with CSM and JSM enabled.
You can see the underlying `jsrun` command generated by `lrun` with the `-v` flag.
Because of the complexity of our `jsrun` calls on Lassen, it may make sense to leave this in and just make note that it is Livermore-specific.

Reply:
> Because of the complexity of our jsrun calls on Lassen, it may make sense to leave this in and just make note that it is Livermore-specific.

That would be my preference: I think it's fine to offer command/batch generators for Livermore-specific systems within this file.